src/apispec/ext/marshmallow/field_converter.py

```diff
@@ -395,15 +395,18 @@ def metadata2properties(self, field, **kwargs):
         """
         # Dasherize metadata that starts with x_
         metadata = {
-            key.replace("_", "-") if key.startswith("x_") else key: value
+            key.replace("_", "-")
+            if isinstance(key, str) and key.startswith("x_")
+            else key: value
             for key, value in field.metadata.items()
         }
 
```

Comment:
I think you could write

```py
        metadata = {
            key.replace("_", "-") if key.startswith("x_") else key: value
            for key, value in field.metadata.items()
            if isinstance(key, str) 
        }
```

to drop non string keys, then remove the test in the second comprehension.

Reply:
You're absolutely right!
done